force-app/main/default/classes/DMLOperations.cls

```diff
@@ -144,9 +178,38 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        //Upsert Account to be associated to an opportunity
+        Account acct = new Account(Name = accountName);
+        List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName WITH SECURITY_ENFORCED Limit 1];
+        if (accounts.isEmpty()) {
+            insert as user acct;
+            } else {
+            acct = accounts[0];
+        }
+        // accounts = [SELECT Id FROM Account WHERE Name = :accountName WITH SECURITY_ENFORCED Limit 1];
+        Id acctId = acct.ID;
+      
         // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        // Create an Opportunity with the given name
+        // Add the Opportunity to the list
+
+        List<Opportunity> listOppt = [SELECT Id, Name FROM Opportunity WHERE Name in : oppNames WITH SECURITY_ENFORCED];
+        if (listoppt.isEmpty()) {
```

Comment:
If you put all of your opportunities into a Map, you can use one for loop to iterate over oppNames and then check if your map contains the oppName in the key.